src/fsharp/FSharp.Core/local.fs

```diff
@@ -125,7 +125,42 @@ module internal List =
                 let cons = freshConsNoTail x
                 chooseToFreshConsTail cons f t
                 cons
-            
+                  
+    let inline groupBy (comparer:IEqualityComparer<'SafeKey>) (keyf:'T->'SafeKey) (getKey:'SafeKey->'Key) (list: 'T list) =
```

Comment:
Actually I reran the tests and inline seems to not make any difference. I've removed it.